Don't overrun output buffer in utf8_ascii() if source string contains embedded NUL
--- mpg123-1.12.4/src/metaprint.c.orig	2010-08-22 14:09:17.000000000 +0200
+++ mpg123-1.12.4/src/metaprint.c	2010-10-03 09:39:15.851740205 +0200
@@ -335,7 +335,7 @@
 	if(!mpg123_resize_string(dest, dlen+1)){ mpg123_free_string(dest); return; }
 	/* Just ASCII, we take it easy. */
 	p = dest->p;
-	for(spos=0; spos < source->fill; ++spos)
+	for(spos=0; (spos < source->fill) && (source->p[spos] != 0); ++spos)
 	{
 		/* UTF-8 continuation byte 0x10?????? */
 		if((source->p[spos] & 0xc0) == 0x80) continue;
